tests/internal/test_database_monitoring.py

```diff
@@ -114,3 +117,24 @@ def test_dbm_propagation_full_mode():
 
     # ensure that dbm tag is set (required for full mode)
     assert dbspan.get_tag(_database_monitoring.DBM_TRACE_INJECTED_TAG) == "true"
+
+
+def test_default_sql_injector(caplog):
```

Comment:
can we create a test case for when the dbm comment would throw a bytes encoding error?

this case would have caught the `dbm_comment = ""` "bug"

is this even possible?

Reply:
decided it is not worth testing

Reply:
@ZStriker19 and I discussed this, it is going to be hard to get `dbm_comment` where it can raise a `UnicodeDecodeError`. The best option we have is probably having a separate file with a different file encoding so we can put non-utf-8 characters into `dbm_comment`, but then try to encode as `utf-8`.

but this is a lot of hoops to jump through to test this.

an alternative could be to mock over `dbm_comment.encode` and force the error as a side effect but... then is that even a good test scenario?



I think as long as the file encoding is `utf-8` or `ascii` then we shouldn't need the try/except around `dbm_comment.encode("utf-8")` since it _shouldnt_ fail, but keeping the try/except as a fail-safe is 👍🏻